src/components/Card.js

```diff
@@ -1,53 +1,70 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { isLiked, name, link, _id },
+    cardSelector,
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButtonClick,
+    removeLikeButton
+  ) {
+    this._isLiked = isLiked;
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleLikeButtonClick = handleLikeButtonClick;
+    this._removeLikeButton = removeLikeButton;
   }
 
   _setEventListeners() {
     this._cardElement
       .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleLikeButton();
+        if (this._isLiked) {
+          this._removeLikeButton(this._id);
+          this._cardElement
+            .querySelector(".card__like-button")
+            .classList.toggle("card__like-button_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.